Source/WebKit/Platform/cocoa/MaterialHostingSupport.swift

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !os(visionOS) && canImport(SwiftUI, _version: 7.0.27)
+
+#if canImport(UIKit)
+@_spi(Private) @_spi(ForUIKitOnly) internal import SwiftUI
+#else
+@_spi(Private) @_spi(ForAppKitOnly) internal import SwiftUI
+#endif
+
+#if canImport(UIKit)
+
+struct MaterialHostingContentViewWrapper: UIViewRepresentable {
+    let contentView: UIView
+
+    func makeUIView(context: Context) -> UIView {
+        contentView
+    }
+
+    func updateUIView(_ uiView: UIView, context: Context) { }
+}
+
+#endif
+
+@objc(WKMaterialHostingSupport)
+class MaterialHostingSupport: NSObject {
+    fileprivate static let materialHostingContentLayerKey = "_materialHostingContentLayerKey"
+
+    @objc
+    class func createHostingLayer() -> CALayer {
+        let contentLayer = CALayer()
+
+        let hostingLayer = CAHostingLayer(rootView: AnyView(_CALayerView(type: CALayer.self) { layer in
```

Comment:
why does this need the AnyView again?

Reply:
Because `.materialEffect` returns `some View`, and the `rootView` is updated later.

Reply:
Got rid of `AnyView`!